wls-common/src/main/java/org/jboss/arquillian/container/wls/ShrinkWrapUtil.java

```diff
@@ -106,22 +112,93 @@ public static URL toURL(final Descriptor descriptor)
     */
    public static File toFile(final Archive<?> archive)
    {
-      // create a random named temp file, then delete and use it as a directory
-      try
-      {
-         File root = File.createTempFile("arquillian", archive.getName());
-         root.delete();
-         root.mkdirs();
-         
-         File deployment = new File(root, archive.getName());
-         deployment.deleteOnExit();
-         archive.as(ZipExporter.class).exportTo(deployment, true);
-         return deployment;
-      }
-      catch (Exception e) 
+      return toFile(archive, false);
+   }
+
+   /**
+    * Creates a tmp folder and exports the file. Returns the URL for that file location.
+    *
+    * @param archive Archive to export
+    * @param exploded Specifies, whether to explode the archive after creation
+    * @return
+    */
+   public static File toFile(final Archive<?> archive, final boolean exploded)
+   {
+       // create a random named temp file, then delete and use it as a directory
+       try
+       {
+           File root = File.createTempFile("arquillian", archive.getName());
+           root.delete();
+           root.mkdirs();
+
+           File deployment = new File(root, archive.getName());
+           deployment.deleteOnExit();
+           archive.as(ZipExporter.class).exportTo(deployment, true);
+           return exploded ? Unzipper.unzip(deployment, new File(new File(root, "exploded"), archive.getName()))
+                           : deployment;
+       }
+       catch (Exception e)
+       {
+           throw new RuntimeException("Could not export deployment to temp", e);
+       }
+   }
+
+    /**
+     * Simple unzip implementation used for extracting contents of deployment archive.
+     */
+   private static class Unzipper

```

Comment:
Using the ExplodedExporter that comes with ShrinkWrap is probably a easier solution

```java
archive.as(ExplodedExporter.class).exportTo...
```